core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/YugabyteDefaultLoadBalancingPolicy.java

```diff
@@ -0,0 +1,194 @@
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.core.metadata.NodeState;
+import com.datastax.oss.driver.api.core.session.Request;
+import com.datastax.oss.driver.api.core.session.Session;
+import com.datastax.oss.driver.api.core.tracker.RequestTracker;
+import com.datastax.oss.driver.internal.core.loadbalancing.BasicLoadBalancingPolicy;
+import com.datastax.oss.driver.internal.core.util.ArrayUtils;
+import com.datastax.oss.driver.internal.core.util.collection.QueryPlan;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Map;
+import java.util.Queue;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.atomic.AtomicLongArray;
+import java.util.function.Predicate;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class YugabyteDefaultLoadBalancingPolicy extends BasicLoadBalancingPolicy implements RequestTracker {
+
+  private static final Logger LOG = LoggerFactory.getLogger(YugabyteDefaultLoadBalancingPolicy.class);
+
+  private volatile DistanceReporter distanceReporter;
+  private volatile Predicate<Node> filter;
+  private volatile String localDc;
+
+  protected final CopyOnWriteArraySet<Node> localDCliveNodes = new CopyOnWriteArraySet<>();
+  protected final Map<Node, AtomicLongArray> responseTimes = new ConcurrentHashMap<>();
+
+  public YugabyteDefaultLoadBalancingPolicy(DriverContext context, String profileName) {
+    super(context, profileName);
+  }
+
+  @Override
+  public void init(@NonNull Map<UUID, Node> nodes, @NonNull DistanceReporter distanceReporter) {
+    this.distanceReporter = distanceReporter;
+    localDc = discoverLocalDc(nodes).orElse(null);
+    filter = createNodeFilter(localDc, nodes);
+
+    for (Node node : nodes.values()) {
+      isLiveNode(node);
+    }
+  }
+
+  @NonNull
+  @Override
+  public Queue<Node> newQueryPlan(@Nullable Request request, @Nullable Session session) {
+    // Take a snapshot since the set is concurrent:
+    Object[] currentNodes = null;
+
+    if (!StringUtils.isBlank(localDc)) {
+
+      currentNodes = localDCliveNodes.toArray();
+      // if there are no healthy nodes in the localDC, then consider nodes from other
+      // DCs.
+      if (currentNodes.length == 0) {
+
+        LOG.trace("[{}] No nodes available in Local DC {}, falling back on to liveNodes", logPrefix, localDc);
+        currentNodes = liveNodes.toArray();
+      }
+
+    } else {
+      currentNodes = liveNodes.toArray();
+    }
+
+    LOG.trace("[{}] Round-robing the {} avaiable nodes", logPrefix, currentNodes.length);
+
+    // Round-robin the remaining nodes
+    ArrayUtils.rotate(currentNodes, 0, currentNodes.length, roundRobinAmount.getAndUpdate(INCREMENT));
+
+    return new QueryPlan(currentNodes);
+  }
```

Comment:
For YB, the local DC should come into picture only for reads at CL.ONE. But from my reading of the driver code we don't have any idea what kind of query this queryPlan is for, do we? If we can figure this out can we limit it to selects only?

Also shouldn't the rotate be guarded by CL like in the current PartitionAwarePolicy implementation? This is also only for reads.